test/unit/module/real/core/logical_andnot.cpp

```diff
@@ -39,18 +38,28 @@ EVE_TEST_TYPES( "Check return types of eve::logical_andnot(simd)"
 EVE_TEST( "Check behavior of eve::logical_andnot(simd)"
         , eve::test::simd::all_types
         , eve::test::generate (eve::test::logicals(0, 3)
-                              , eve::test::logicals(1, 2))
+                              , eve::test::logicals(1, 2)
+                              , eve::test::randoms(0, 2))
         )
-<typename M>(M const& l0, M const& l1)
+<typename M, typename T>(M const& l0, M const& l1, T const & a0)
 {
   using eve::detail::map;
-  using v_t = eve::element_type_t<M>;
+  using l_t = eve::element_type_t<M>;
 
-    TTS_EQUAL(eve::logical_andnot(l0, true), eve::false_(eve::as<M>()));
-    TTS_EQUAL(eve::logical_andnot(true, l1), map([](auto e) -> v_t { return  !e; }, l1));
-    TTS_EQUAL(eve::logical_andnot(false, l1), eve::false_(eve::as<M>()));
-    TTS_EQUAL(eve::logical_andnot(l0, false), map([](auto e) -> v_t { return  e; }, l0));
-    TTS_EQUAL(eve::logical_andnot(l0, l1), map([](auto e, auto f) -> v_t { return  e && !f; }, l0, l1));
-    TTS_EQUAL(eve::logical_andnot(l0, l1.get(0)), map([&](auto e) -> v_t { return  e && !l1.get(0); }, l0));
-    TTS_EQUAL(eve::logical_andnot(l0.get(0), l1), map([&](auto f) -> v_t { return  l0.get(0) && !f; }, l1));
+  TTS_EQUAL(eve::logical_andnot(l0, true), eve::false_(eve::as<M>()));
+  TTS_EQUAL(eve::logical_andnot(true, l1), map([](auto e) -> l_t { return  !e; }, l1));
+  TTS_EQUAL(eve::logical_andnot(false, l1), eve::false_(eve::as<M>()));
+  TTS_EQUAL(eve::logical_andnot(l0, false), map([](auto e) -> l_t { return  e; }, l0));
+  TTS_EQUAL(eve::logical_andnot(l0, l1), map([](auto e, auto f) -> l_t { return  e && !f; }, l0, l1));
+  TTS_EQUAL(eve::logical_andnot(l0, l1.get(0)), map([&](auto e) -> l_t { return  e && !l1.get(0); }, l0));
+  TTS_EQUAL(eve::logical_andnot(l0.get(0), l1), map([&](auto f) -> l_t { return  l0.get(0) && !f; }, l1));
+  using v_t = eve::element_type_t<T>;
+  using d_t = eve::detail::downgrade_t<v_t>;
+  auto da0 = eve::convert(a0, eve::as<d_t>());
+  using dl_t = eve::as_logical_t<d_t>;
+  std::cout << "l1 " << l1 << std::endl;
```

Comment:
spurrious std::cout

Reply:
ok but I said it was still just a try on one function 
I will do the other logical_xxx

Reply:
i made the corrections